adding parameters

AnnAnnFryingPan opened the pull request.

AnnAnnFryingPan committed:
Started adding params to AURN_data_Download, added gitIgnore

AnnAnnFryingPan committed:
Removed data_url variable as not used

AnnAnnFryingPan committed:
Passed value for useful_num_years to be 0.8*len(years)

AnnAnnFryingPan committed:
Started testing/debugging when only a subset of sites

douglowe commented on AURN_Data_Download/AURN_download.py:
```diff
@@ -626,53 +665,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
-    hourly_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe= hourly_dataframe.rename(columns={'siteID':'SiteID'})
-    
+    #daily_dataframe = (site_list,metadata,years,data_path,save_to_csv)
```
We should delete this commented out code completely - it's just me being untidy.

AnnAnnFryingPan replied:
Have removed that comment.

douglowe commented on AURN_Data_Download/AURN_download.py:
```diff
@@ -626,53 +665,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
     # postprocessing the data set, to get daily data
-    daily_dataframe = postprocess_organisation(hourly_dataframe,emep_dataframe,stations,site_list,
-                                                   use_emep_data,impute_values)
+    useful_num_years = 0.8*len(years)
```
I think we should have optional parameters for defining 'useful_num_years' and 'min_years' - and then move the code for calculating defaults for these based on the number of years chosen by the user into the routine itself.

AnnAnnFryingPan replied:
I've got this running, by setting the station_distances loop to go up to a maximum of the number of station distances, as it was going over that number before. I've also set a maximum value for station_number (but I don't think that was what broke it, but has potential to, if number of sites request by user is <5.

AnnAnnFryingPan replied:
But this needs your testing.

AnnAnnFryingPan replied:
Also, I'm not used to setting parameters for commandline apps, and followed your lead with the emep processing code.
I'm not sure I process the boolean arguments properly. I'll check/test these.

douglowe replied:
These look fine to me. In addition I've added a check in the reading in of the parameters, so that useful_num_years is always greater or equal to min_years.

douglowe reviewed (commented):
It's mostly looking good - I just think we could do with a better fix than my cludge for the useful_num_years (and min_years) control of which stations are used as reference stations, and for which we are including in our final dataset.

The choice of how to calculate these might depend the results from the tests we make for the reliability of the imputation methods.

AnnAnnFryingPan committed:
Removed daily_dataframe setting (was commented out)

AnnAnnFryingPan committed:
Added min_years and useful_num_years as user parameters

AnnAnnFryingPan committed:
Replaced \t with \n

AnnAnnFryingPan committed:
Returned to \t, added extra tabs to output for readability

douglowe commented on AURN_Data_Download/AURN_download.py:
```diff
@@ -626,53 +688,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
+            print('\t{} has {} positive values'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]>0.0])))
+            print('\t{} has {} NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc].isna()])))
+            print('\t{} has {} negative or zero values that will be replaced with NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]<=0.0])))
+            hourly_dataframe.loc[hourly_dataframe[spc]<=0.0, 'spc'] = np.nan
```
is this the location of our error - spc shouldn't be in a string?

AnnAnnFryingPan replied:
Oh yes!!

AnnAnnFryingPan committed:
Fixed bug where hourly_dataframe.loc[..] 2nd param is variable not string.

AnnAnnFryingPan committed:
Added verbose output for debugging. Set a maximum value for station_number. Set station_distances loop to go up to a maximum of the number of station distances

AnnAnnFryingPan committed:
Fixed Verbose by setting it as integer

AnnAnnFryingPan committed:
For boolean user input arguments, copied way done in emep_extract. Needs testing

douglowe committed:
Added a couple of notes on default values to the help interface.

douglowe committed:
Added check to ensure that "useful_num_years" >= "min_years"

Making sure that the reference stations for our imputation
process (defined as the stations with more data than "useful_num_years")
will always have at least the same amount of data as our
cut off point for real data for the stations we will finally output.

douglowe committed:
Adding check that useful_num_years >= min_years to defaults too.

douglowe committed:
Making the Path usage more consistent.

I've made all files Path objects, which has
eliminated the need for the use of os.path
(so that we're consistent with packages used).

douglowe committed:
Made treatment of EMEP file more robust.

Replaced checks for the existence of the dataframe
with checks to make sure it is not empty, and
changed the behaviour when there is no EMEP data to
simply create an empty dataframe.

douglowe committed:
Changed processing of VERBOSE flag.

Changed check to see if it exists, and if so
then we use the maximum value of verbose and 0.

douglowe commented:
I've made a few changes, to ensure internal logic, and also to enable the use of emep data files.
